examples/notebooks/beam-ml/rag_usecase/opensearch_rag_pipeline.ipynb

```diff
@@ -0,0 +1,1779 @@
+    "from apache_beam.transforms.enrichment import Enrichment\n",
+    "from apache_beam.ml.transforms.embeddings.huggingface import SentenceTransformerEmbeddings\n",
+    "import tempfile\n",
+    "import json\n",
+    "\n",
+    "from opensearchpy import OpenSearch\n",
+    "from opensearch_connector import *\n",
+    "from opensearch_enrichment import *\n",
+    "from chunks_generation import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "3d274890-4e6b-4a3d-b682-9fc6e21e5cca",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'2.56.0'"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#To check beam version installed \n",
+    "beam.__version__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "84d2e424-2747-421b-8ab0-8416992536b6",
+   "metadata": {},
+   "source": [
+    "Next we load the Opensearch credentials that will be needed in access during following steps. It reads credentials from environment. Please replace the values in the file before running this."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "5948f80b-65fa-4416-b7be-803c56477b72",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from dotenv import load_dotenv\n",
+    "\n",
+    "load_dotenv('credentials.env')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "394e2ebf-9553-4464-b6a3-1386ef57f24b",
+   "metadata": {},
+   "source": [
+    "# Reading json data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "006fdc19-e16a-4142-9653-b0c0a7e63b8e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('hf_small_wikipedia.json', 'r') as j:\n",
+    "     contents = json.loads(j.read())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "213fb96c-a641-4688-b083-96fd762608aa",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "list"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "type(contents)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bdc5047c-8539-4381-84fd-5660f9829a10",
+   "metadata": {},
+   "source": [
+    "# Creating a Search Index¶\n",
+    "### Next few cells will show how to specify and create a search index in OpenSearch vector DB. \n",
+    "\n",
+    "Set some constants for defining our index like the distance metric and the index name.\n",
+    "\n",
+    "Define the index schema with OpenSearchSearch fields.\n",
+    "\n",
+    "Create the index.\n",
+    "\n",
+    "Index creation is neeeded only once."
```

Comment:
These steps can feel kind of disjointed written as-is, it's a little more friendly to the user to turn these into a more coherent paragraph or make them into a bulleted list of steps.

Reply:
Reworded